matter_server/client/models/node.py

```diff
@@ -270,18 +261,27 @@ def get_cluster(
         """
         return self.endpoints[endpoint].get_cluster(cluster)
 
-    @property
-    def name(self) -> str | None:
-        """Return friendly name for this node."""
-        if self.root_device_type_instance is None:
-            return None
-        return cast(
-            str,
-            self.root_device_type_instance.endpoint.get_attribute_value(
-                None,
-                Clusters.BasicInformation.Attributes.NodeLabel,
-            ),
-        )
+    def update(self, node_data: MatterNodeData) -> None:
+        """Update MatterNode from MatterNodeData."""
+        self.node_data = node_data
+        # collect per endpoint data
+        endpoint_data: dict[int, dict[str, Any]] = {}
+        for attribute_path, attribute_data in node_data.attributes.items():
+            endpoint_id = int(attribute_path.split("/")[0])
+            endpoint_data.setdefault(endpoint_id, {})
```

Comment:
maybe better to check if key exists instead of using `setdefault`, otherwise you're going to create and throw away a lot of empty dicts.